[NNC] added skeleton for refactoring

Chillee opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #56456 Added matmul/unified dtypes
* #55372 [NNC] Move computeNOperand operations over
* **#55371 [NNC] added skeleton for refactoring**

Differential Revision: [D27616418](https://our.internmc.facebook.com/intern/diff/D27616418)

Chillee committed:
added skeleton for refactoring

[ghstack-poisoned]

Chillee committed:
Update on "added skeleton for refactoring"




[ghstack-poisoned]

Chillee committed:
Update on "[NNC] added skeleton for refactoring"




[ghstack-poisoned]

ZolotukhinM reviewed (approved):
Looks good overall! I left several minor remarks below:

Chillee commented on torch/csrc/jit/tensorexpr/kernel.cpp:
```diff
@@ -422,6 +467,26 @@ std::vector<ExprHandle> TensorExprKernel::inferSizesForValue(
+  //   // the operator-specific lowering code.
+  //   return IntImm::make(0);
+  } else {
+    throw unsupported_dtype();
```
It's the same one that's used right below it (in the original `constant`).

Chillee commented on torch/csrc/jit/tensorexpr/kernel.cpp:
```diff
@@ -422,6 +467,26 @@ std::vector<ExprHandle> TensorExprKernel::inferSizesForValue(
+  } else {
+    throw unsupported_dtype();
+  }
+  assert(false);
```
I think I had a lint error at some point ... which is why I did this.

Chillee committed:
Update on "[NNC] added skeleton for refactoring"




[ghstack-poisoned]

Chillee commented on torch/csrc/jit/tensorexpr/kernel.cpp:
```diff
@@ -665,6 +779,30 @@ Tensor* TensorExprKernel::computeTwoOperand(
 
+Tensor* TensorExprKernel::computeTwoOperand(
+    const std::string& name,
+    const std::vector<ArgValue> inputValues,
```
I think I'm going to refrain from this, as I may want to combine these various `computeNOperand` functions in a follow up PR.

Chillee committed:
Update on "[NNC] added skeleton for refactoring"


Differential Revision: [D27616418](https://our.internmc.facebook.com/intern/diff/D27616418)

[ghstack-poisoned]

Chillee committed:
Update on "[NNC] added skeleton for refactoring"


Differential Revision: [D27616418](https://our.internmc.facebook.com/intern/diff/D27616418)

[ghstack-poisoned]

Chillee committed:
Update on "[NNC] added skeleton for refactoring"


Differential Revision: [D27616418](https://our.internmc.facebook.com/intern/diff/D27616418)

[ghstack-poisoned]
